examples/Basic.hs

```diff
@@ -9,15 +9,64 @@
 
 {-# LANGUAGE OverloadedStrings   #-}
 {-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TemplateHaskell #-}
 module Main where
 
 import qualified Data.Text                as T
 import qualified Network.Wai.Handler.Warp as W
+import qualified Network.Wai              as W
 import           UnliftIO                 (liftIO)
 import qualified UnliftIO.Exception       as E
+import qualified Data.Aeson               as A
+import           Data.Aeson.TH            (deriveToJSON)
+import           Data.Aeson.Casing        (aesonDrop, camelCase)
+import qualified Network.HTTP.Types       as HT
+import qualified Data.ByteString.Char8    as B8
+import           Data.ByteString          (ByteString)
 
 import           Webby
 
+-- | Data types and combinators for structured logging
+data ApiInfo = ApiInfo { apiMethod      :: T.Text
+                       , apiPath        :: T.Text
+                       , apiQueryParams :: T.Text
+                       }
+               deriving (Eq, Show)
+
+$(deriveToJSON (aesonDrop 3 camelCase) ''ApiInfo)
+
+
+data AppLogInfo = AppLogInfo { aliAppName   :: T.Text
+                             , aliApi       :: ApiInfo
+                             , aliException :: T.Text
+                             }
+                  deriving (Eq, Show)
+
+$(deriveToJSON (aesonDrop 3 camelCase) ''AppLogInfo)
+
+fromBSToText :: ByteString -> T.Text
+fromBSToText = T.pack . B8.unpack
+
+appExceptionHandler :: T.Text -> WebbyExceptionHandler appEnv
+appExceptionHandler appName =
+    mkWebbyExceptionHandler $
+    E.Handler $
+    \(exn :: E.SomeException) -> do
```

Comment:
I have a doubt this will catch the control flow exceptions as well (like FinishThrown, etc). Can you verify?

Reply:
I have verified this by running `examples/Basic.hs`. The reason it works is the user provided exception handler is installed before the exception handler responsible for webby handler's control flow.